sdk/peering/Azure.ResourceManager.Peering/tests/Scenario/PeeringServiceTests.cs

```diff
@@ -0,0 +1,119 @@
+        {
+            _resourceGroup = await CreateResourceGroup();
+        }
+
+        [RecordedTest]
+        public async Task CreateOrUpdate()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            var peeringService = await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            ValidatePeeringService(peeringService);
+            Assert.AreEqual(peeringServiceName, peeringService.Data.Name);
+        }
+
+        [RecordedTest]
+        public async Task Exist()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            bool flag = await _peeringServiceCollection.ExistsAsync(peeringServiceName);
+            Assert.IsTrue(flag);
+        }
+
+        [RecordedTest]
+        public async Task Get()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            var peeringService = await _peeringServiceCollection.GetAsync(peeringServiceName);
+            ValidatePeeringService(peeringService);
+            Assert.AreEqual(peeringServiceName, peeringService.Value.Data.Name);
+        }
+
+        [RecordedTest]
+        public async Task GetAll()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            var list = await _peeringServiceCollection.GetAllAsync().ToEnumerableAsync();
+            Assert.IsNotEmpty(list);
+            ValidatePeeringService(list.First(item => item.Data.Name == peeringServiceName));
+        }
+
+        [RecordedTest]
+        public async Task Delete()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            var peeringService = await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            bool flag = await _peeringServiceCollection.ExistsAsync(peeringServiceName);
+            Assert.IsTrue(flag);
+
+            await peeringService.DeleteAsync(WaitUntil.Completed);
+            flag = await _peeringServiceCollection.ExistsAsync(peeringServiceName);
+            Assert.IsFalse(flag);
+        }
+
+        [RecordedTest]
+        public async Task AddTag()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            var peeringService = await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            await peeringService.AddTagAsync("addtagkey", "addtagvalue");
+
+            peeringService = await _peeringServiceCollection.GetAsync(peeringServiceName);
+            KeyValuePair<string, string> tag = peeringService.Data.Tags.Where(tag => tag.Key == "addtagkey").FirstOrDefault();
```

Comment:
I think it's better to add a line to verify that the peering service only has one tag here.